plugin-cgroupv2/src/k8s/utils.rs

```
@@ -362,6 +360,40 @@ pub async fn get_pod_name(
     Ok(("".to_string(), "".to_string(), "".to_string()))
 }
 
+/// Retrieves the k8s API token using either a kubectl command
+/// or by reading  the service account token's file.
+fn get_token(token_retrieval: String) -> Option<String> {
+    match token_retrieval.as_str() {
+        "kubectl" => {
+            let Ok(output) = Command::new("kubectl")
+                .args(["create", "token", "alumet-reader", "-n", "alumet"])
+                .output()
+            else {
+                return None;
```

The function should return `anyhow::Result` instead of silently ignoring errors